### PR TITLE
Expose a query endpoint for latest published valset

### PR DESCRIPTION
**Pull request by ToasterTheBrave**
# Related Github tickets

- https://github.com/VolumeFi/paloma/issues/783

# Background

Pigeon currently gets this information by querying the compass contract. The problem is, when there is a just-in-time valset update, pigeon will end up using an old valset because the new valset hasn't been committed yet.  We need the newest valset that may still be pending on the evm chain.

Now, we can query paloma instead of the evm contract.  Paloma knows what the latest published valset is, even if it hasn't yet been committed.

# Testing completed

- [x] test coverage exists or has been added/updated
- [x] tested in a private testnet

# Breaking changes

- [x] I have checked my code for breaking changes
- [x] If there are breaking changes, there is a supporting migration.

### CODE REVIEW

File: x/valset/client/cli/query_get_latest_published_snapshot.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import (
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/palomachain/paloma/x/valset/types"
+	"github.com/spf13/cobra"
+)
+
+func CmdGetLatestPublishedSnapshot() *cobra.Command {
+	cmd := &cobra.Command{
```
Comment (byte-bandit) [inline x/valset/client/cli/query_get_latest_published_snapshot.go:11]: we dont need the cli command for pigeons to pick up this info, but it won't hurt either :)

**Review by byte-bandit — APPROVED** [review]
lgtm

File: x/valset/keeper/grpc_query_get_latest_published_snapshot.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	snapshot, err := k.GetLatestSnapshotOnChain(ctx, req.ChainReferenceID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryGetLatestPublishedSnapshotResponse{
+		Snapshot: snapshot,
+	}, nil
+}
```
Comment (byte-bandit) [inline x/valset/keeper/grpc_query_get_latest_published_snapshot.go:26]: Nitpicking, low hanging fruit to improve test coverage here.

Comment (ToasterTheBrave) [reply x/valset/keeper/grpc_query_get_latest_published_snapshot.go:26]: I'm working on the tests right now

File: x/valset/client/cli/query_get_latest_published_snapshot.go (hunk 1 above)
Comment (ToasterTheBrave) [reply x/valset/client/cli/query_get_latest_published_snapshot.go:11]: Super easy to include the CLI command.  I didn't see any harm in it and it makes it easier to debug

File: x/valset/keeper/grpc_query_get_latest_published_snapshot.go (hunk 2 above)
Comment (ToasterTheBrave) [reply x/valset/keeper/grpc_query_get_latest_published_snapshot.go:26]: Tests added

**Pull request merged**